pkg/sql/drop_table.go

```diff
@@ -411,6 +417,34 @@ func (p *planner) unsplitRangesForTable(ctx context.Context, tableDesc *tabledes
 					return err
 				}
 			}
+
+			rest, tableID, indexID, err := p.ExecCfg().Codec.DecodeIndexPrefix(desc.StartKey.AsRawKey())
+			if err != nil {
+				continue
+			}
+			if tableID != uint32(tableDesc.GetID()) {
+				return errors.Errorf("expected table %d, got table ID %d", tableDesc, tableID)
+			}
+			mappedID, ok := indexIDMapping[descpb.IndexID(indexID)]
+			if !ok {
+				continue
+			}
+			indexPrefix := p.ExecCfg().Codec.IndexPrefix(tableID, uint32(mappedID))
+			mappedSplitPoints = append(mappedSplitPoints, append(indexPrefix, rest...))
+		}
+		// Downsample, remap to new table, split.
+		// TODO(nvanbenschoten): base this on num nodes or something.
+		newRanges := 64
```

Comment:
What would be the downside of mapping all of the range splits? I guess the point is that if there was a huge amount of data in the table that was being truncated, but much of the ranges were not active, it would be a waste to recreate those ranges...